substrate/bin/node/cli/src/chain_spec.rs

```diff
@@ -314,31 +316,87 @@ pub fn testnet_genesis(
 	let (initial_authorities, endowed_accounts, stakers) =
 		configure_accounts(initial_authorities, initial_nominators, endowed_accounts, STASH);
 
-	// Todo: After #7748 is done, we can refactor this to avoid
-	// calling into the native runtime.
-	kitchensink_runtime::genesis_config_presets::kitchensink_genesis(
-		initial_authorities
+	let validator_count = initial_authorities.len();
+	let minimum_validator_count = validator_count;
+
+	let collective = collective(&endowed_accounts);
+
+	serde_json::json!({
+		"balances": {
+			"balances": endowed_accounts.iter().cloned().map(|x| (x, ENDOWMENT)).collect::<Vec<_>>()
+		},
+		"session": {
+			"keys": initial_authorities
 			.iter()
 			.map(|x| {
 				(
 					x.0.clone(),
 					// stash account is controller
 					x.0.clone(),
-					session_keys(
+					session_keys_json(
 						x.2.clone(),
 						x.3.clone(),
 						x.4.clone(),
 						x.5.clone(),
 						x.6.clone(),
 						x.7.clone(),
-					),
+					)
 				)
 			})
-			.collect(),
-		root_key,
-		endowed_accounts,
-		stakers,
-	)
+			.collect::<Vec<_>>()
+		},
+		"elections": {
+			"members": collective.iter().cloned().map(|member| (member, STASH)).collect::<Vec<_>>(),
+		},
+		"technicalCommittee": {
+			"members": collective,
+		},
+		"staking": {
+			"validatorCount": validator_count,
+			"minimumValidatorCount": minimum_validator_count,
+			"invulnerables": initial_authorities
+				.iter()
+				.map(|x| x.0.clone())
+				.collect::<Vec<_>>(),
+			"stakers": stakers,
+		},
+		"sudo": {
+			"key": root_key,
+		},
+		"revive": {
+			"mappedAccounts": endowed_accounts.iter().filter(|x| ! is_eth_derived(x)).cloned().collect::<Vec<_>>()
+		}
+	})
+}
+
+/// Creates the session keys as defined by the runtime.
+fn session_keys_json(
+	grandpa: GrandpaId,
+	babe: BabeId,
+	im_online: ImOnlineId,
+	authority_discovery: AuthorityDiscoveryId,
+	mixnet: MixnetId,
+	beefy: BeefyId,
+) -> serde_json::Value {
+	serde_json::json!({
+		"authority_discovery": authority_discovery,
+		"babe": babe,
+		"beefy": beefy,
+		"grandpa": grandpa,
+		"im_online": im_online,
+		"mixnet": mixnet
+	})
+}
+
+/// Extract some accounts from endowed to be put into the collective.
+fn collective(endowed: &[AccountId]) -> Vec<AccountId> {
+	const MAX_COLLECTIVE_SIZE: usize = 50;
+	let endowed_accounts_count = endowed.len();
+	endowed
+		.iter()
+		.take((endowed_accounts_count.div_ceil(2)).min(MAX_COLLECTIVE_SIZE))
+		.cloned()
+		.collect()
 }
```

Comment:
This is a duplicate from the runtime's genesis presets. It might make sense to extract a `genesis-build-helper` crate, which is used by the runtime's genesis presets, but can also be used here.

Reply:
Does this [comment](https://github.com/paritytech/polkadot-sdk/pull/8323/files#r2060778015) make sense, or is there anything preventing us from such refactor?

Reply:
Answered at your comment.